lib/Routes/Opencast/UserRoles.php

```diff
@@ -180,6 +168,36 @@ public function __invoke(Request $request, Response $response, $args)
                     $roles[$course_id . '_Learner'] = $course_id . '_Learner';
                 }
 
+                // get all videos the user has permissions on excluding videos from courses with write perm
```

Comment:
Please write a better comment description, as it is a bit not clear!

Reply:
```suggestion
                // Get all videos permissions of the user except those for videos to which the user has already write
                // permissions through course memberships and course playlists,  in order to reduce the number of roles.
```

I hope this comment is clearer.